### ISSUE TITLE
(🐞) F821 false positive with `__future__.annotations`

### ISSUE THREAD

**Issue by KotlinIsland** [astral-sh/ruff#6112]
```py
from __future__ import annotations

from typing import Literal

E = Literal["P"]  # F821 Undefined name `P`
```

python 3.11
ruff 0.0.280

**Comment by charliermarsh** [comment]
This is fixed on main, thank you for reporting, sorry about that.
